shared/src/androidMain/kotlin/io/customer/shared/tracking/queue/QueueTimer.kt

```diff
@@ -0,0 +1,102 @@
+package io.customer.shared.tracking.queue
+
+import android.os.CountDownTimer
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobDispatcher
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.TimeUnit
+import io.customer.shared.work.runOnMain
+
+internal actual fun getQueueTimer(
+    logger: Logger,
+    jobExecutor: JobExecutor,
+): QueueTimer = AndroidQueueTimer(
+    logger = logger,
+    jobExecutor = jobExecutor,
+)
+
+/**
+ * [QueueTimer] implementation that uses Android [CountDownTimer] to run timer countdowns.
+ */
+private class AndroidQueueTimer(
+    private val logger: Logger,
+    override val jobExecutor: JobExecutor,
+) : QueueTimer, JobDispatcher {
+    @Volatile
+    private var countdownTimer: CountDownTimer? = null
+    private val timerAlreadyScheduled: Boolean
+        get() = countdownTimer != null
+
+    override fun onCoroutineFailed(exception: Throwable) {
+        countdownTimer = null
+        logger.error(message = "Timer failed with error ${exception.message}")
+    }
+
+    override fun schedule(force: Boolean, duration: TimeUnit.Seconds, block: () -> Unit) {
```

Comment:
```suggestion
    override fun schedule(cancelPrevious: Boolean, duration: TimeUnit.Seconds, block: () -> Unit) {
```

`force` doesn't describe well what the behavior of that parameter is for.